src/bundle.c

```diff
@@ -426,126 +433,157 @@ int remove_bundle(const char *bundle_name)
+		if (!is_tracked_bundle(*bundles)) {
+			fprintf(stderr, "Warning: Bundle \"%s\" does not seem to be installed\n", *bundles);
+			ret = EBUNDLE_NOT_TRACKED;
+			bad++;
+			goto out_free_curl;
+		}
 
-	/* load all submanifest minus the one to be removed */
-	current_mom->submanifests = recurse_manifest(current_mom, subs, NULL, false);
-	if (!current_mom->submanifests) {
-		fprintf(stderr, "Error: Cannot load MoM sub-manifests\n");
-		ret = ERECURSE_MANIFEST;
-		goto out_free_mom;
-	}
+		current_mom = load_mom(current_version, false, mix_exists);
+		if (!current_mom) {
+			fprintf(stderr, "Unable to download/verify %d Manifest.MoM\n", current_version);
+			ret = EMOM_NOTFOUND;
+			bad++;
+			goto out_free_curl;
+		}
 
-	struct list *reqd_by = NULL;
-	required_by(&reqd_by, bundle_name, current_mom, 0);
-	if (reqd_by != NULL) {
-		struct list *iter;
-		char *bundle;
-		iter = list_head(reqd_by);
-		fprintf(stderr, "Error: bundle requested to be removed is required by the following bundles:\n");
-		printf("format:\n");
-		printf(" # * is-required-by\n");
-		printf(" #   |-- is-required-by\n");
-		printf(" # * is-also-required-by\n # ...\n\n");
-		while (iter) {
-			bundle = iter->data;
-			iter = iter->next;
-			fprintf(stderr, "%s", bundle);
+		if (!search_bundle_in_manifest(current_mom, *bundles)) {
+			fprintf(stderr, "Bundle name is invalid, aborting removal\n");
+			ret = EBUNDLE_REMOVE;
+			bad++;
+			goto out_free_mom;
 		}
 
-		list_free_list(reqd_by);
-		ret = EBUNDLE_REMOVE;
-		goto out_free_mom;
-	}
+		/* load all tracked bundles into memory */
+		read_subscriptions(&subs);
 
-	current_mom->files = files_from_bundles(current_mom->submanifests);
+		/* popout the bundle to be removed from memory */
+		ret = unload_tracked_bundle(*bundles, &subs);
+		if (ret != 0) {
+			bad++;
+			goto out_free_mom;
+		}
 
-	current_mom->files = consolidate_files(current_mom->files);
+		set_subscription_versions(current_mom, NULL, &subs);
 
-	/* Now that we have the consolidated list of all files, load bundle to be removed submanifest*/
-	ret = load_bundle_manifest(bundle_name, subs, current_version, &bundle_manifest);
-	if (ret != 0 || !bundle_manifest) {
-		fprintf(stderr, "Error: Cannot load %s sub-manifest (ret = %d)\n", bundle_name, ret);
-		goto out_free_mom;
-	}
+		/* load all submanifests minus the one to be removed */
+		current_mom->submanifests = recurse_manifest(current_mom, subs, NULL, false);
+		if (!current_mom->submanifests) {
+			fprintf(stderr, "Error: Cannot load MoM sub-manifests\n");
+			ret = ERECURSE_MANIFEST;
+			bad++;
+			goto out_free_mom;
+		}
 
-	/* deduplication needs file list sorted by filename, do so */
-	bundle_manifest->files = list_sort(bundle_manifest->files, file_sort_filename);
-	deduplicate_files_from_manifest(&bundle_manifest, current_mom);
+		/* check if bundle is required by another installed bundle */
+		struct list *reqd_by = NULL;
+		required_by(&reqd_by, *bundles, current_mom, 0);
```

Comment:
How on earth did I let required_by get into the code base - I must have been asleep!